src/ZoneServer/World/Actors/Characters/Character.cs

```diff
@@ -720,7 +722,16 @@ public void Heal(float hpAmount, float spAmount)
 			if (hpAmount == 0 && spAmount == 0)
 				return;
 
-			this.ModifyHpSafe(hpAmount, out var hp, out var priority);
+			float healingReduction = 0;
```

Comment:
This is a bit of a tricky change. I did tell you we need to add the buff check were it applies, and this method _is_ called `Heal`, and it is used by potions and the Heal skill, but it may also be used by NPCs, commands, etc, where we expect it to actually heal for the amount specified, regardless of buffs. For example, imagine a Healer NPC that uses `Heal()`, which in turn calls this method. Visiting the Healer would then not fully heal you while this debuff is active, which would be weird. This needs to be documented at the very least, and we could also modify `Heal()` to give a bit more healing, to compensate for potential debuffs, or perhaps change it to do its own full heal, without relying on this method.

Reply:
I understand your point but I am not sure how to implement this in a way that it can be used universally.
Should I be creating a issue related to it? Is the best I can do.

Reply:
The easiest solution for now would be to just document it, and perhaps fix the full heal in some way, so it actually fully heals. I'll look into it during the merge.